### PR TITLE
c++ mode: update rules for embedded R chunks to allow for no space between *** and R

### PR DESCRIPTION
**Pull request by jjallaire**
@kevinushey Could you review this? 

One change I'm not 100% certain about is the change to the regex passed to the RCodeModel. I actually changed it back to what it was when we first checked this feature in:

https://github.com/rstudio/rstudio/commit/80a6c25cd96f92edf7932a6bdafe6d61b8c44ee2

The way I was reading the regex you had in there was /***<space> (then any characters) <eol>. That doesn't seem right to me (as we require an r or R) but then it occurred to me that you may have changed it to that for a reason I don't understand. This checkin reverts to the original regex, let me know if you think that is okay.

### CODE REVIEW

**Comment by kevinushey** [comment]
The capturing regex was inserted in this commit:

https://github.com/rstudio/rstudio/commit/afdc9dab1bab031469f0957b6edc11556b856d92

It looks like I initially added the capturing regex to allow chunk labels to be properly inferred for R code chunks, but then ended up not using it and just assigning the name `(R Code Chunk)` irrespective of the regex capture. (Previously, the `getChunkLabel` function was failing as the regex used defined no capture group; note that this _does_ exist in e.g. RMarkdown):

https://github.com/rstudio/rstudio/blob/master/src/gwt/acesupport/acemode/rmarkdown.js#L44-L45

In short -- there should be a single capturing group for chunk labels; however, since we no longer use this in C++ mode (for R chunks) it may be unnecessary.
